### PR TITLE
implemented base regitration endpoint

### PR DESCRIPTION
**Pull request by Maxim-Gumeniuk**

### CODE REVIEW

**Comment by Maxim-Gumeniuk** [comment]
@SpaceHolderdJs

**Comment by Maxim-Gumeniuk** [comment]
[sergienko339@gmail.com](mailto:sergienko339@gmail.com)

File: src/controllers/auth/registr/index.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { userModel } from "@/db/users";
+import { Request, Response } from "express"
+
+
+const registr = async (req: Request, res: Response) => {
```
Comment (SpaceHolderdJs) [inline src/controllers/auth/registr/index.ts]: register

File: src/controllers/auth/registr/index.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        res.send({
+            newUser,
+            message: 'new user created'
+        });
+    } catch(e) {
+        console.log(e);
+
+        const error = new Error(e as string);
+
+        res.send(error.message)
```
Comment (SpaceHolderdJs) [inline src/controllers/auth/registr/index.ts]: Error status is needed to represent additional info (403, 500, 404 ect)

File: src/controllers/auth/registr/index.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        
+    }
+}
+
+export const registrationController = {
+    registr,
```
Comment (SpaceHolderdJs) [inline src/controllers/auth/registr/index.ts]: register

File: src/db/users.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { ENVVARIABLES } from "@/env-variables"
+import { DB } from "@/constructors/usersDb.ts";
+import { emailValidations } from "@/helpers/credentials-validation";
```
Comment (SpaceHolderdJs) [inline src/db/users.ts]: Please never use custom validation, it`s done here 

1. https://www.npmjs.com/package/express-validator#documentation

File: src/db/users.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+const { MONGOSTRING } = ENVVARIABLES;
+
+export const usersDb = new DB(MONGOSTRING!, {
+    email: { 
+        type: String, 
+        unique : true, 
+        required : true,
+        validate: [emailValidations, 'it`s should be email address']
```
Comment (SpaceHolderdJs) [inline src/db/users.ts]: Good practice 👍🏻

File: src/server.ts [hunk 6]
```diff
@@ -1,15 +1,31 @@
...
+import cors from "cors";
+
+import { ENVVARIABLES } from "@/env-variables";
+import { authRouter } from "./routes/auth-router";
+import { connectUsersDb } from "./db/users";
 
 const app = express();
 
-const { PORT } = ENVVARIABLES;
+const { PORT, CLIENT_APP_URL } = ENVVARIABLES;
```
Comment (SpaceHolderdJs) [inline src/server.ts:10]: Perfect! 👍🏻

File: src/types/routes/registration/index.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export const enum Registration  {
```
Comment (SpaceHolderdJs) [inline src/types/routes/registration/index.ts]: Better to name it like `RegistrationLinks`

**Review by SpaceHolderdJs — COMMENTED** [review]
So far looks great 💯

**Pull request merged**